index.js

```diff
@@ -233,12 +233,33 @@ var handleCloudWatch = function(event, context) {
   var region = event.Records[0].EventSubscriptionArn.split(":")[3];
   var subject = "AWS CloudWatch Notification";
   var alarmName = message.AlarmName;
-  var metricName = message.Trigger.MetricName;
+  var trigger = message.Trigger;
+  var alarmExpression;
+  if (typeof trigger.MetricName === "undefined") {
+    // This is a CloudWatch metric math alarm. Instead of a MetricName and
+    // statistic there is an array of Metrics where the first element is the
+    // math expression.
+
+    // first build a dictionary mapping metric ids to metric name and statistic
+    var sourceMetrics = {};
+    for (const metric of trigger.Metrics.slice(1)) {
+      sourceMetrics[metric.Id] = metric.MetricStat.Stat + ":"
+        + metric.MetricStat.Metric.MetricName;
+    }
+
+    // now replace each instance of the metric id in the alarm expression
+    alarmExpression = trigger.Metrics[0].Expression;
+    for (var metricid in sourceMetrics) {
+      alarmExpression = alarmExpression.replace(new RegExp(metricid,"g"),sourceMetrics[metricid]);
```

Comment:
If `metric.Id` is not intended to be a regular expression itself, it should probably be escaped. As is, a malicious user can submit a custom, unsanitized regular expression as input, which [may produce](https://www.owasp.org/index.php/Regular_expression_Denial_of_Service_-_ReDoS) [undesirable effects](https://en.wikipedia.org/wiki/ReDoS) [if we parse it](https://lgtm.com/rules/2025670640/).

```javascript
function sanitizeStringForRegExp(str) {
  return str.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
}
alarmExpression = alarmExpression.replace(
  new RegExp(sanitizeStringForRegExp(metricid), 'g'),
  sourceMetrics[metricid]
);
```

Source: [Stack Overflow](https://stackoverflow.com/questions/1144783/how-to-replace-all-occurrences-of-a-string)

Reply:
incorporated escape-string-regexp package which implements the equivalent of sanitizeStringForRegExp above.